go/vt/sqlparser/sql.y

```diff
@@ -8497,6 +8513,7 @@ non_reserved_keyword:
 | SNAPSHOT
 | SOME %prec ANY_SOME
 | SQL
+| SQL_BUFFER_RESULT
```

Comment:
`HIGH_PRIORITY`, `SQL_SMALL_RESULT` and `SQL_BIG_RESULT` need to be added to this list too.